packages/connect-core/src/async-iterable.ts

```diff
@@ -1126,3 +1126,136 @@ export function makeIterableAbortable<T>(
+  const process = async (payload: IteratorResult<T, undefined>) => {
+    // // If the writer's internal error was set, then reject any attempts at processing a payload.
+    if (error) {
+      return Promise.reject(String(error));
+    }
+    // If there is an iterator resolver then a consumer of the async iterator is waiting on a value.  So resolve that
+    // promise with the new value being sent and return a promise that is immediately resolved
+    if (queueResolve) {
+      queueResolve(payload);
+      queueResolve = undefined;
+      return Promise.resolve();
+    }
+    const elem: QueueElement<T> = {
+      payload,
+    };
+    const prom = new Promise<void>((resolve, reject) => {
+      elem.resolve = resolve;
+      elem.reject = reject;
+    });
+    // Otherwise no one is waiting on a value yet so add it to the queue and return a promise that will be resolved
+    // when someone reads this value
+    queue.push(elem);
+
+    return prom;
+  };
+  const send = async (payload: T) => {
+    return process({ value: payload, done: false });
+  };
+  const close = async () => {
+    return process({ value: undefined, done: true });
+  };
+
+  let closed = false;
+  return {
+    isClosed() {
+      return closed;
+    },
+    async write(payload) {
+      if (closed) {
+        throw new ConnectError("cannot write, already closed");
+      }
+      return send(payload);
+    },
+    async close() {
+      if (closed) {
+        throw new ConnectError("cannot close, already closed");
+      }
+      closed = true;
+      return close();
+    },
+    [Symbol.asyncIterator](): AsyncIterator<T> {
+      return {
+        next: async () => {
+          // If the writer's internal error was set, then reject any attempts at processing a payload.
+          if (error) {
+            return Promise.reject(String(error));
+          }
+          const elem = queue.shift();
+          if (!elem) {
+            // We don't have any payloads ready to be sent (i.e. the consumer of the iterator is consuming faster than
+            // senders are sending).  So return a Promise ensuring we'll resolve it when we get something.
+            return new Promise<IteratorResult<T>>((resolve) => {
+              queueResolve = resolve;
+            });
+          }
+          // Resolve the send promise on a successful send/close.
+          if (elem.resolve) {
+            elem.resolve();
+          }
+          return elem.payload;
+        },
+        throw: async (e: Error) => {
+          error = e;
+          // The reader of this iterator has failed with the given error.  So anything left in the queue should be
+          // drained and rejected with the given error
+          for (const item of queue) {
+            if (item.reject) {
+              item.reject(e);
+            }
+          }
+          queue = [];
+          return new Promise<IteratorResult<T>>((resolve) => {
+            resolve({ value: String(e), done: true });
+          });
```

Comment:
Hm, why return the error as a string here?

Wouldn't a simple `undefined` be more fitting?

```suggestion
          return new Promise<IteratorResult<T>>((resolve) => {
            resolve({ value: undefined, done: true });
          });
```

Reply:
I don't know, I feel like it's nice to have the error information here returned.  That way you know that the reason the writer is returning `done` is because `throw` was invoked with an error.  

I don't feel super strongly about it though.  Just feels more intuitive than an empty string.

Reply:
You are basically doing this:

```ts
async function* gen() {
  try {
    yield 1;
  } catch (e) {
    return String(e)
  }
}
```

I'm not sure there's a good answer here, but this doesn't seem right. Let's just return `undefined`.

Reply:
Sounds good.  Updated.